### PR TITLE
Add matchAnchorWidth to popover

### PR DESCRIPTION
**Pull request by newyork-anthonyng**
This commit is a WIP. 

I will add tests to this if this is the way to go.

### CODE REVIEW

File: packages/boundless-popover/index.js [hunk 1]
```diff
@@ -411,16 +417,34 @@ export default class Popover extends React.PureComponent {
         this.$caret.style[longitudinal ? 'top' : 'left'] = '0px';
     }
 
+    matchAnchorWidth = () => {
+      if (!this.props.matchAnchorWidth) return;
+
+      const anchor = this.props.anchor instanceof HTMLElement
```
Comment (probablyup) [inline packages/boundless-popover/index.js]: Just learned this recently, but `findDOMNode` is actually a no-op if called on a DOM node, so you can remove this ternary and always call it.

File: packages/boundless-popover/index.js [hunk 2]
```diff
@@ -411,16 +417,34 @@ export default class Popover extends React.PureComponent {
...
+                     ? this.props.anchor
+                     : findDOMNode(this.props.anchor);
+      const dialog = this.dialog.$dialog;
+      const anchorWidth = anchor.getBoundingClientRect().width;
```
Comment (probablyup) [inline packages/boundless-popover/index.js]: We compute this in `cacheViewportCartography()` as called by `this.align()`, so if we move the invocation of this function to after the caching is complete, we can avoid doing the measurement work twice :smile:

File: packages/boundless-popover/index.js [hunk 3]
```diff
@@ -411,16 +417,34 @@ export default class Popover extends React.PureComponent {
...
+      dialog.style.width = `${anchorWidth}px`;
+    }
+
     componentDidMount() {
+        this.matchAnchorWidth();
         this.align();
+        window.addEventListener('resize', this.matchAnchorWidth, true);
```
Comment (probablyup) [inline packages/boundless-popover/index.js]: I think we can remove this additional event listener as well by integrating the width matching function into the alignment flow.

**Review by quantizor — CHANGES_REQUESTED** [review]
Great start! Some comments attached.

**Comment by newyork-anthonyng** [comment]
@probablyup Thanks for the comments. I will make the updates and work on adding tests this weekend.

**Comment by newyork-anthonyng** [comment]
I made the changes and added a test for this. Let me know if there's anything else needed :)

**Review by quantizor — APPROVED** [review]

**Comment by probablyup** [comment]
Sorry for the very long delay, this looks excellent.

**Pull request merged**